samples/drivers/spi_flash/sample.yaml

```diff
@@ -18,6 +19,7 @@ tests:
   sample.drivers.spi.flash_dpd:
     tags: spi flash
     filter: dt_compat_enabled("jedec,spi-nor")
+    platform_exclude: hifive_unmatched
```

Comment:
> hifive_unmatched has a jedec,spi-nor flash node, however we have no
SPI driver for the controller so exclude the platform from this sample.

wondering how does it work... sample entry also has `depends_on: spi`, why isn't hifive being excluded?

Reply:
let me take another look at what's going on here w/hifive_unmatched`

Reply:
So this is because the spi device instance isn't created because of how SPI0 is treated.  If you look at `drivers/spi/Kconfig.sifive` there is an `SIFIVE_SPI_0_ROM` option that is enabled which effectively disables the SPI0 controller.